### PR TITLE
CORE-7764: add local request checking to the DE controller

### PR DESCRIPTION
**Pull request by dennisroberts71**

### CODE REVIEW

File: ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java [hunk 1]
```diff
@@ -46,12 +49,30 @@ public String redirectToDe(final HttpServletRequest request) throws MalformedURL
...
+
+    private boolean isLocalRequest(HttpServletRequest req) {
+        IpRanges ipRanges = new IpRanges(localIpRanges);
+        String forwardedFor = req.getHeader("X-Forwarded-For");
+        if (forwardedFor != null && forwardedFor.length() != 0) {
+            return ipRanges.matches(forwardedFor.split("\\s*,\\s*")[0]);
+        } else {
+            return ipRanges.matches(req.getRemoteAddr());
+        }
+    }
```
Comment (psarando) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: This code looks similar to the DeLandingPage#localRequest method.
Could this logic be refactored into a common place?

Comment (dennisroberts71) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: Yeah, I could probably create a static function somewhere that takes a request and a set of IP ranges and checks them.

File: ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java [hunk 2]
```diff
@@ -46,12 +49,30 @@ public String redirectToDe(final HttpServletRequest request) throws MalformedURL
         return "redirect:de/";
     }
 
+    private DiscoveryEnvironmentMaintenance getDeMaintenance() {
+        return new DiscoveryEnvironmentMaintenance(maintenanceFile);
```
Comment (psarando) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: In the current code the `DiscoveryEnvironmentMaintenance` is constructed with `System.getProperty("user.dir") + "/" + maintenanceFile`.
Does just passing `maintenanceFile` work here?

File: ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java (hunk 1 above)
Comment (dennisroberts71) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: Heh, or for that matter, I could probably make it an instance method of IpRanges.

Comment (psarando) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: Good point, I was wondering that myself :)

File: ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java (hunk 2 above)
Comment (dennisroberts71) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java]: Yeah, `user.dir` resolves to the current working directory, so it's effectively the same as long as the path doesn't start with a path separator. When the path does start with a path separator, the current code will treat the path as absolute whereas the previous code will treat it as relative to the current working directory. The drop page code has always used just the maintenance file property name, so I went with that option.

**Comment by psarando** [comment]
My comments were pretty minor, so otherwise this looks good to me :shipit:

**Pull request merged**